### ISSUE TITLE
Collapse animations don't clear the transition queue before beginning

### ISSUE THREAD

**Issue by jdhines** [twbs/bootstrap#20881]
**Use Case**
http://codepen.io/RedOctober13/pen/GjdBKx
When toggling a collapse (particularly one that uses arrow/caret indicators to show open/close, if the button is clicked multiple times quickly, the animation takes its time to finish before allowing any other animations. This means that if you click twice, the icon which flipped down then back up quickly, indicates the collapsible section is closed when it's really open.

Transitions for Collapse should use the `stop()` function before animating.
**Edit** Or, if I'm just way off base, the effect I want is that if I click to collapse while it's still expanding, I want it to stop expanding and start collapsing, and vice versa.

Reported on Mac OS X, Google Chrome 53

**Comment by twbs-lmvtfy** [comment]
Hi @jdhines!

You appear to have posted a live example (http://codepen.io/RedOctober13/pen/GjdBKx.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 18, column 1 thru column 7: End tag for  ```` body ```` seen, but there were unclosed elements.
* line 14, column 1 thru column 43: Unclosed element ```` div ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by jdhines** [comment]
Unclosed element has been fixed.

**Comment by cvrebert** [comment]
> the stop() function

Not sure exactly what you're referring to here.

**Comment by jdhines** [comment]
The jQuery one: http://api.jquery.com/stop/
Perhaps `clearQueue ()` would work too, but chaining in `stop()` seems to be used more.

**Comment by mxviteri** [comment]
@jdhines Glad you brought this up. I currently use Bootstrap v3.3.7 in an enterprise application and I see the same issue. In the past I've avoided this and used jQuery slideUp() and slideDown() along with stop() to achieve the desired result.
ex. $this.parents(".panel-body").stop().slideUp();

**Comment by cvrebert** [comment]
Bootstrap does not use jQuery animations. It uses CSS animations only.

**Comment by jdhines** [comment]
Thanks @mxviteri. That's hard to do in my project with Ember.js since I'm not triggering the transitions from the controller.
@cvrebert That's cool. Is there a way to short-circuit CSS animations so an animation can be stopped in order to begin a new one?

Right now I'm solving this by just dispensing with the arrow icon that switches between side-pointing and down-pointing, so it doesn't look out of sync.

**Comment by mdo** [comment]
> Is there a way to short-circuit CSS animations so an animation can be stopped in order to begin a new one?

I don't believe this is possible in CSS, so closing as a can't/won't fix.
